lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -151,6 +168,97 @@ public boolean matches(DictionaryDescriptor dictionary)
         return effectivePredicateDomain == null || effectivePredicateMatches(effectivePredicateDomain, dictionary);
     }
 
+    private Optional<Collection<Object>> extractDiscreteValues(ValueSet valueSet)
+    {
+        if (!valueSet.isDiscreteSet()) {
+            return valueSet.tryExpandRanges(domainCompactionThreshold);
+        }
+
+        return Optional.of(valueSet.getDiscreteSet());
+    }
+
+    @Override
+    public boolean matches(ColumnDescriptor columnDescriptor, ColumnChunkMetaData columnMetaData, BloomFilterStore bloomFilterStore)
+    {
+        requireNonNull(bloomFilterStore, "bloomFilterStore is null");
+
+        if (effectivePredicate.isNone()) {
+            return false;
+        }
+        Map<ColumnDescriptor, Domain> effectivePredicateDomains = effectivePredicate.getDomains()
+                .orElseThrow(() -> new IllegalStateException("Effective predicate other than none should have domains"));
+
+        if (!effectivePredicateDomains.containsKey(columnDescriptor)) {
+            return true;
+        }
+
+        Domain effectivePredicateDomain = effectivePredicateDomains.get(columnDescriptor);
+        // the bloom filter bitset contains only non-null values so isn't helpful
+        if (effectivePredicateDomain.isNullAllowed()) {
+            return true;
+        }
+        Optional<BloomFilter> bloomFilterOptional = bloomFilterStore.readBloomFilter(columnMetaData);
+        if (!bloomFilterOptional.isPresent()) {
+            return true;
+        }
+        BloomFilter bloomFilter = bloomFilterOptional.get();
+
+        Optional<Collection<Object>> discreteValues = extractDiscreteValues(effectivePredicateDomain.getValues());
+        if (discreteValues.isEmpty()) {
+            // values are not discrete, so bloom filter isn't helpful
+            return true;
+        }
+        return discreteValues.get().stream().anyMatch(value -> {
+            Type type = effectivePredicateDomain.getType();
+            return mightInBloomFilter(bloomFilter, value, type);
+        });
+    }
+
+    /**
+     * Check if the predicateValue might be in the bloomfilter
+     *
+     * @param bloomFilter parquet bloomfilter.
+     * @param predicateValue effective discrete predicate value.
+     * @param sqlType Type that contains information about the type schema from connector's metadata
+     * @return true if the predicateValue might be in the bloomfilter, false if the predicateValue absolutely is not in the bloomfilter
+     */
+    private static boolean mightInBloomFilter(BloomFilter bloomFilter, Object predicateValue, Type sqlType)
+    {
+        try {
+            long hashValue = getBloomFilterHash(bloomFilter, predicateValue, sqlType);
+            return bloomFilter.findHash(hashValue);
+        }
+        catch (UnsupportedOperationException unsupportedOperationException) {
```

Comment:
Let's avoid the use of exception for control flow here.
Return `true` for unsupported data types and get rid of the try/catch

Reply:
done.